### PR TITLE
Fallback to CPU when datasource v2 enables RuntimeFiltering

### PR DESCRIPTION
**Pull request by wbo4958**
This PR is to fix https://github.com/NVIDIA/spark-rapids/issues/3093

### CODE REVIEW

**Comment by wbo4958** [comment]
build

File: shims/spark320/src/main/scala/com/nvidia/spark/rapids/shims/spark320/Spark320Shims.scala [hunk 1]
```diff
@@ -483,15 +485,40 @@ class Spark320Shims extends Spark32XShims {
...
+            TypeSig.DECIMAL_64).nested(),
+          TypeSig.all),
+        (p, conf, parent, r) => new SparkPlanMeta[BatchScanExec](p, conf, parent, r) {
+          override val childScans: scala.Seq[ScanMeta[_]] =
+            Seq(GpuOverrides.wrapScan(p.scan, conf, Some(this)))
+
+          override def tagPlanForGpu(): Unit = {
+            // Only Scan with SupportsRuntimeFiltering can support DPP.
+            if (!p.runtimeFilters.isEmpty) {
+              willNotWorkOnGpu("The DPP has not been supported")
```
Comment (revans2) [inline shims/spark320/src/main/scala/com/nvidia/spark/rapids/shims/spark320/Spark320Shims.scala:502]: Can we update the comment to be a little more clear? Perhaps something like.

"Runtime filtering (DPP) on datasource V2 is not supported"

File: shims/spark320/src/main/scala/com/nvidia/spark/rapids/shims/spark320/Spark320Shims.scala [hunk 2]
```diff
@@ -483,15 +485,40 @@ class Spark320Shims extends Spark32XShims {
...
   }
 
   override def getScans: Map[Class[_ <: Scan], ScanRule[_ <: Scan]] = Seq(
     GpuOverrides.scan[ParquetScan](
       "Parquet parsing",
       (a, conf, p, r) => new ScanMeta[ParquetScan](a, conf, p, r) {
-        override def tagSelfForGpu(): Unit = GpuParquetScanBase.tagSupport(this)
+        override def tagSelfForGpu(): Unit = {
+          GpuParquetScanBase.tagSupport(this)
+          if (a.isInstanceOf[SupportsRuntimeFiltering]) {
```
Comment (revans2) [inline shims/spark320/src/main/scala/com/nvidia/spark/rapids/shims/spark320/Spark320Shims.scala:518]: Could we add a comment here that we are being overly cautious and that Parquet does not support this yet?

This goes for the other versions too.

**Review by revans2 — APPROVED** [review]
Looks good to me. Just a few nits

**Comment by wbo4958** [comment]
build

**Comment by wbo4958** [comment]
build

**Review by revans2 — DISMISSED** [review]

File: shims/spark320/src/main/scala/com/nvidia/spark/rapids/shims/spark320/Spark320Shims.scala [hunk 3]
```diff
@@ -510,8 +538,14 @@ class Spark320Shims extends Spark32XShims {
     GpuOverrides.scan[OrcScan](
       "ORC parsing",
       (a, conf, p, r) => new ScanMeta[OrcScan](a, conf, p, r) {
-        override def tagSelfForGpu(): Unit =
+        override def tagSelfForGpu(): Unit = {
           GpuOrcScanBase.tagSupport(this)
+          // we are being overly cautious and that Orc does not support this yet
+          if (a.isInstanceOf[SupportsRuntimeFiltering]) {
```
Comment (gerashegalov) [inline shims/spark320/src/main/scala/com/nvidia/spark/rapids/shims/spark320/Spark320Shims.scala:544]: here and elsewhere: Does calling tagSupport(this) on L542 make sense if  this is true?

Comment (wbo4958) [reply shims/spark320/src/main/scala/com/nvidia/spark/rapids/shims/spark320/Spark320Shims.scala:544]: Per my understanding. Plugin is trying best to explain **all the reasons** that won't run on GPU at a time.

**Comment by wbo4958** [comment]
build

**Review by firestarman — APPROVED** [review]

**Pull request merged**